core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -420,14 +399,21 @@ public short abortTransaction(final long transactionID) throws Exception {
          if (useTlog) {
             tLog.putSingleRecord(transactionID, -1, "ABORTED", ts.getParticipatingRegions(), false);
          }
-      } catch(Exception e) {
-         LOG.error("Returning from HBaseTxClient:abortTransaction, txid: " + transactionID + " tLog.putRecord: EXCEPTION");
+      } catch(IOException e) {
+         LOG.error("Returning from HBaseTxClient:abortTransaction, txid: " + transactionID + " tLog.putRecord: EXCEPTION", e);
          return TransReturnCode.RET_EXCEPTION.getShort();
       }
 
       if ((stallWhere == 1) || (stallWhere == 3)) {
          LOG.info("Stalling in phase 2 for abortTransaction");
-         Thread.sleep(300000); // Initially set to run every 5 min
+         boolean loopBack = true;
+         do {
+            try {
+               Thread.sleep(300000); // Initially set to run every 5 min
+            } catch(InterruptedException ie) {
+               loopBack = true;
```

Comment:
Should there be a loopBack = false after Thread.sleep in the case where an exception occurred but then the operation succeeded?